### PR TITLE
pkg/ansible,pkg/scaffold/ansible; enables leader election for ansible operators

### PR DESCRIPTION
**Pull request by mhrivnak**
Leader election has been enabled for a while for go-based operators. This uses the same approach and very similar code.

### CODE REVIEW

File: pkg/ansible/run.go [hunk 1]
```diff
@@ -56,7 +61,31 @@ func Run(flags *aoflags.AnsibleOperatorFlags) {
 		log.Fatal(err)
 	}
 
-	printVersion()
+	name, found := os.LookupEnv(k8sutil.OperatorNameEnvVar)
+	if !found {
+		log.Fatal("OPERATOR_NAME environment variable not set")
```
Comment (hasbro17) [inline pkg/ansible/run.go]: Just a nit but I thought we were using logr for all the operator logs. We're using logrus here.

Comment (mhrivnak) [reply pkg/ansible/run.go]: Gotcha. I presume then we need to change this whole file, which is maybe better-done as a follow-up PR?

Comment (hasbro17) [reply pkg/ansible/run.go]: Yep we can change this in a separate PR.

**Comment by mhrivnak** [comment]
I removed the readiness probe, so this is good to go.

**Comment by mhrivnak** [comment]
> Worried that this is actually on always and not on by default. I don't see a way to turn of the lock is that correct?

Correct. That is also currently the case for go-based operators. We don't have a way to turn it off at runtime. It would be a reasonable thing to add as a command line arg or similar, but that goes back to the discussion we're having on #889 about how to change runtime settings.

**Pull request merged**